src/api/plugin-api.ts

```diff
@@ -87,6 +88,8 @@ export class DataviewApi {
     public widget = Widgets;
     /** Re-exporting of luxon for people who can't easily require it. Sorry! */
     public luxon = Luxon;
+    /** Programmatic extensions to tweak dataview behavior */
+    public extensionRegistry = extensionRegistry;
```

Comment:
The reason to expose `extensionRegistry` instead of a direct API call `addCustomSpecialTaskFieldExtractor` is to have a central place for such overrides. I would anticipate more of them to be added later.

The long-term vision is for `dataview` to be one and only query engine for any other obsidian plugin. For instance, I would like to completely replace the built-in task querying functionality in obsidian-tasks with dataview.

Reply:
Another potential use-case for the extension registry is to allow for custom indexers - should resolve #1803 without the need to add the canvas support directly to the core.

Overall, the philosopy is to maintain a robust lean core and let third party extensions to take care of new extra features.

WDYT?